### ISSUE TITLE
Carbon throws error when PostgreSQL has removed trailing milliseconds

### ISSUE THREAD

**Issue by JohnOakwood** [laravel/framework#28832]
- Laravel Version: 5.8.8
- PHP Version: 7.3.6
- Database Driver & Version: (PostgreSQL) 9.6.11
- Carbon Version: 2.19.2

### Description:
**TLDR;** When trying to get `created_at` field from the model, Carbon throws `InvalidArgumentException`, because milliseconds are missing from the timestamp field. Carbon waits for the `Y-m-d H:i:s.u` format, but PostgreSQL has stripped milliseconds.

If PostgreSQL timestamp value is saved in DB as follows: `2019-01-12 09:00:00.000000`, then when querying data PostgreSQL removes those (`.000000`) milliseconds and returns value as follows `2019-01-12 09:00:00` (this doesn't happen in standard cases, where milliseconds isn't `.000000`. 

Date format for Carbon is set to `Y-m-d H:i:s.u`, but Carbon receives value without milliseconds and throws following error `at Carbon::rawCreateFromFormat('Y-m-d H:i:s.u', '2019-01-12 09:00:00', null)`.

How to resolve this case? Is there an option to set a fallback date format? It would be cool if Carbon could handle this case on his own.

### Steps To Reproduce:
1) Create a PostgreSQL table with timestamp precision 6
2) Save data with zero milliseconds (`2019-01-12 09:00:00.000000`)
3) Set models date format to `Y-m-d H:i:s.u`
4) Query the model
5) Try to get value from model `$myCoolModel->created_at`
6) See the error
7) cry

**Comment by driesvints** [comment]
Which version of Carbon are you using? Does upgrading to the latest version of Laravel and Carbon fixes your problem?

**Comment by staudenmeir** [comment]
This is a known issue without a good solution. The workaround is defining custom accessors:

```php
class User extends Model
{
    protected $dateFormat = 'Y-m-d H:i:s.u';

    public function getCreatedAtAttribute($value)
    {
        $format = Str::contains($value, '.') ? $this->dateFormat : 'Y-m-d H:i:s';

        return Date::createFromFormat($format, $value);
    }

    public function getUpdatedAtAttribute($value)
    {
        $format = Str::contains($value, '.') ? $this->dateFormat : 'Y-m-d H:i:s';

        return Date::createFromFormat($format, $value);
    }
}
```

I think we should consider adding a `$fallbackDateFormat` property to handle cases like this.

The issue is even more annoying with `toArray()` where accessors don't work: #27169

**Comment by JohnOakwood** [comment]
> Which version of Carbon are you using? Does upgrading to the latest version of Laravel and Carbon fixes your problem?

Hi, my Carbon version is: `2.19.2` I have the latest version, but it does not address the issue.

**Comment by JohnOakwood** [comment]
> This is a known issue without a good solution. The workaround is defining custom accessors:
> 
> ```
> class User extends Model
> {
>     protected $dateFormat = 'Y-m-d H:i:s.u';
> 
>     public function getCreatedAtAttribute($value)
>     {
>         $format = Str::contains($value, '.') ? $this->dateFormat : 'Y-m-d H:i:s';
> 
>         return Date::createFromFormat($format, $value);
>     }
> 
>     public function getUpdatedAtAttribute($value)
>     {
>         $format = Str::contains($value, '.') ? $this->dateFormat : 'Y-m-d H:i:s';
> 
>         return Date::createFromFormat($format, $value);
>     }
> }
> ```
> 
> I think we should consider adding a `$fallbackDateFormat` property to handle cases like this.
> 
> The issue is even more annoying with `toArray()` where accessors don't work: #27169

Hi, thank you for the response, Your solution works without problems, thank you for that!

I just was hoping that there is some "cleaner" fix, it's not super efficient to add workaround like this to every model. 

Still, thank you for this fix, it will have to do for now.

**Comment by driesvints** [comment]
Glad you found a fix for now. Closing this as a duplicate of https://github.com/laravel/framework/issues/27169
